File: src/main/java/com/experimento/copilot/entity/Ride.java
package com.experimento.copilot.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@Entity(name = "RIDE")
@AllArgsConstructor
@NoArgsConstructor
public class Ride implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(targetEntity = Vehicle.class)
    @JoinColumn(name = "VEHICLE_ID", nullable = false)
    private Vehicle vehicle;

    @ManyToOne(targetEntity = User.class)
    @JoinColumn(name = "USER_ID", nullable = false)
    private User rider;

    @Column(name = "DATE")
    private LocalDateTime date;


}

File: src/Caronas.Domain/Ride.cs
namespace Caronas.Domain
{
    public class Ride
    {
        public string? Id { get; set; }
        public string? VehicleId { get; set; }
        public Vehicle? Vehicle { get; set; }
        public string? UserId { get; set; }
        public User? User { get; set; }
        public DateTime Date { get; set; }
    }
}